File: challenges/9-likes/go/Uriel-David/likes.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {
	fmt.Print("Digite os nomes no formato ['Nome1', 'Nome2', ...]: ")
	reader := bufio.NewReader(os.Stdin)
	input, _ := reader.ReadString('\n')

	input = strings.ReplaceAll(input, "[", "")
	input = strings.ReplaceAll(input, "]", "")
	input = strings.ReplaceAll(input, "'", "")
	input = strings.TrimSpace(input)

	likes := strings.Split(input, ",")
	formattedLikes := formatLikes(likes)
	fmt.Println(formattedLikes)
}

func formatLikes(names []string) string {
	lenNames := len(names)

	if names[0] == "" {
		return "Ninguém curtiu isso"
	} else if lenNames == 1 {
		return fmt.Sprintf("%s curtiu isso", names[0])
	} else {
		phrase := names[0]
		for i := 1; i < lenNames-1; i++ {
			phrase += fmt.Sprintf(", %s", names[i])
		}
		phrase += fmt.Sprintf(" e%s curtiram isso", names[lenNames-1])

		return phrase
	}
}

File: challenges/9-likes/dart/lokoinsanus/likes.dart
main(List<String> args) {
  String argsString = args[0].substring(1, args[0].length - 1);
  List<String> argsList = argsString
      .split(', ')
      .map((element) => element.replaceAll("'", ''))
      .toList();
  print(likes(argsList));
}

likes(List<String> cont) {
  var length = cont.length;
  if (cont[0] == '') {
    return "Ninguém curtiu isso";
  } else if (length == 1) {
    return "${cont[0]} curtiu isso";
  } else if (length == 2) {
    return "${cont[0]} e ${cont[1]} curtiram isso";
  } else if (length == 3) {
    return "${cont[0]}, ${cont[1]} e ${cont[2]} curtiram isso";
  }
  return "${cont[0]}, ${cont[1]} e ${length - 2} pessoas curtiram isso";
}
